resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ResourceTest.java

```diff
@@ -1127,6 +1131,7 @@ protected void setUp() throws Exception {
 
 	@Override
 	protected void tearDown() throws Exception {
+		 restorePreviousStateOfJobManager();
```

Comment:
I would actually expect the `JobManager` to always be running (before a test) and have it running afterwards again. Otherwise a test will have side effects on others. Shouldn't we thus fail the test throwing an exception in case the `JobManager` was not reenabled (like we do after improper cleanup)?

Reply:
I changed so the previous state is still restored but I also failed the test. This will not only improve isolation but also let us know which tests are not cleaning up properly.

Reply:
I do not see any scenario where the "old" may ever be different from unsuspended. There are only two reasons why that can happen:
1. The JobManager was suspended before the test, i.e., some other test suspended it and did not resume. That is a symptom of improper test isolation and should either bei covered by an assertion at the beginning of the test or not be handled by passing that faulty state to the next test.
2. The test has already suspended the JobManager in its setup logic (as the old state is stored after a test might have disabled the JobManager during its setup instead of storing it before). But hat is exactly what the added functionality is supposed to avoid.

So I would simply resume the `JobManager` here and throw an assertion error if the test has not resumed it by itself.

Reply:
>I do not see any scenario where the "old" may ever be different from unsuspended. 

If a previous test which does **not** extend `ResourceTest` stops the `JobManager` then the initial state is "suspended". I didn't want to make any assumptions about which kind of tests ran before or what the "right" state is, I just wanted to make sure every test (that extends `ResourceTest`) cleans up accordingly.

Reply:
That is exactly the first scenario I have described. But if that happens, the initial state for the test is faulty, so it does not make sense to me to restore a faulty state and pass it to the next test.

A test should always assume a "clean state" when starting. And in case it is in doubt that the system is in such a state, it should check and notify about that state rather than dealing with it and passing that state to the next test. So to catch the scenario you described, you would have to assert that the `JobManager` is not suspended when the test starts.

Reply:
> That is exactly the first scenario I have described.

I know, I just wanted to emphasize the fact about not making assumptions about which one is the "right" initial state :-) . But I see your point: tests that need to suspend the `JobManager` are the exception; most of the tests run under the assumption that the `JobManager` is **not** suspended. Considering "suspended" as the "faulty state" should be fine.

I changed the code in bcf4c13d8d026c83ba3f2ea62c8715d72c9531f3

Reply:
The documentation is quite clear which state of the `JobManager` is to be considered exceptional ;-)